File: 9/Problem4.c
#include<stdio.h>

int MultDigit(int iNo)
{
    int iCount = 0;
    int iDigit = 0;
    int iMult = 1;

    while(iNo != 0)
    {
        iDigit = iNo % 10;
        if( iDigit != 0 )
        {
            iMult = iMult * iDigit;
        }
        iNo = iNo / 10;
    }
        
    
    return iMult;
}
int main()
{
    int iValue = 0;
    int iRet = 0;
    printf("Enter the number : \n");
    scanf("%d",&iValue);
    iRet = MultDigit(iValue);
    printf("%d\n",iRet);
    return 0;
}

File: 20/problem4.java
import java.util.*;

class DigitX
{
    public int LargestDigit(int iNo)
    {
        int iDigit = 0;
        int iLargest = 0;

        while(iNo != 0)
        {
            iDigit = iNo % 10 ;
            if(iDigit > iLargest)
            {
                iLargest = iDigit;
            }
            iNo = iNo/10;
        }
        return iLargest;
    }
}
class problem4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        DigitX dobj = new DigitX();

        int iValue = 0, iRet = 0;

        System.out.println("Enter the number : ");
        iValue = sobj.nextInt();

        iRet = dobj.LargestDigit(iValue);
        System.out.println("Largest digit is : "+iRet);

    }
}